C/common/reading_set.cpp

```diff
@@ -192,6 +192,55 @@ ReadingSet::append(const vector<Reading *>& readings)
 	}
 }
 
+/**
+* Deep copy a set of readings to this reading set.
+*/
+bool
+ReadingSet::copy(const ReadingSet& src)
+{
+   vector<Reading *> readings;
+   bool copyResult = true;
+   try
+   {
+	   // Iterate over all the readings in ReadingSet
+	   for ( auto &reading : src.getAllReadings())
+	   {
+		   std::string assetName = reading->getAssetName();
+		   std::vector<Datapoint *> dataPoints;
+		   // Iterate over all the datapoints associated with one reading
+		   for ( auto &dp : reading->getReadingData())
+		   {
+			   std::string dataPointName  = dp->getName();
+			   DatapointValue dv = dp->getData();
+			   dataPoints.emplace_back(new Datapoint(dataPointName, dv));
+			  
+		   }
+		   Reading *in = new Reading(assetName, dataPoints);
+		   readings.emplace_back(in);
+	   }
+   }
+   catch (std::bad_alloc& ex)
+   {
+		Logger::getLogger()->error("Insufficient memory, failed while copying %d reading from ReadingSet, %s ",readings.size()+1, ex.what());
+		copyResult = false;
+		readings.clear();
+   }
+   catch (std::exception& ex)
+   {
+	   Logger::getLogger()->error("Unknown exception, failed while copying %d reading from ReadingSet, %s ",readings.size()+1, ex.what());
+	   copyResult = false;
```

Comment:
same here, memory on heap handling

Reply:
Vector of readings is created on stack so it will taken care during stack unwinding, No need to delete separately